File: app/src/test/java/com/mss/imagesearcher/di/MockModule.java
package com.mss.imagesearcher.di;

import androidx.annotation.NonNull;

import com.mss.imagesearcher.presenter.detail.DetailPresenter;
import com.mss.imagesearcher.presenter.imagelist.ImageListPresenter;
import com.mss.imagesearcher.presenter.info.InfoPresenter;
import com.mss.imagesearcher.view.helpers.ImageLoader;

import org.mockito.Mockito;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
class MockModule {

    @Singleton
    @Provides
    @NonNull
    ImageLoader provideImageLoader() {
        return Mockito.mock(ImageLoader.class);
    }

    @Singleton
    @Provides
    @NonNull
    ImageListPresenter provideImageListPresenter() {
        return Mockito.mock(ImageListPresenter.class);
    }

    @Singleton
    @Provides
    @NonNull
    DetailPresenter provideDetailPresenter() {
        return Mockito.mock(DetailPresenter.class);
    }

    @Singleton
    @Provides
    @NonNull
    InfoPresenter provideInfoPresenter() {
        return Mockito.mock(InfoPresenter.class);
    }
}

File: app/src/test/java/com/mss/imagesearcher/di/MockModule.kt
package com.mss.imagesearcher.di

import com.mss.imagesearcher.presenter.detail.DetailPresenter
import com.mss.imagesearcher.presenter.imagelist.ImageListPresenter
import com.mss.imagesearcher.presenter.info.InfoPresenter
import com.mss.imagesearcher.view.helpers.ImageLoader
import dagger.Module
import dagger.Provides
import org.mockito.Mockito
import javax.inject.Singleton

@Module
internal class MockModule {

    @Singleton
    @Provides
    fun provideImageLoader(): ImageLoader {
        return Mockito.mock(ImageLoader::class.java)
    }

    @Singleton
    @Provides
    fun provideImageListPresenter(): ImageListPresenter {
        return Mockito.mock(ImageListPresenter::class.java)
    }

    @Singleton
    @Provides
    fun provideDetailPresenter(): DetailPresenter {
        return Mockito.mock(DetailPresenter::class.java)
    }

    @Singleton
    @Provides
    fun provideInfoPresenter(): InfoPresenter {
        return Mockito.mock(InfoPresenter::class.java)
    }
}
